book_source/03_intermediate_users_guide/01_pecan_xml.Rmd

```diff
@@ -208,26 +500,27 @@ Only if this section is defined an ensemble analysis is done.
-Available sampling methods for `parameters` can be found in the documentation of the `get.ensemble.samples` function in `PEcAn.utils` package.
-For the cases where we need simulations with a predefined set of parameters, met and initial condition we can use the restart argument. Restart needs to be a list with name tags of runid, inputs, new.params (parameters), new.state (initial condition), ensemble.id (ensemble ids), start.time and stop.time.
+Each piece in the sampling space can potentially have a method tag and a parent tag. Method refers to the sampling method and parent refers to the cases where we need to link the samples of two components. When no tag is defined for one component, one sample will be generated and used for all the ensembles. This allows for partitioning/studying different sources of uncertainties. For example, if no met tag is defined then, one met path will be used for all the ensembles and as a result the output uncertainty will come from the variability in the parameters. At the moment no sampling method is implemented for soil and vegetation.
+Available sampling methods for `parameters` can be found in the documentation of the `PEcAn.utils::get.ensemble.samples` function.
+For the cases where we need simulations with a predefined set of parameters, met and initial condition we can use the restart argument. Restart needs to be a list with name tags of `runid`, `inputs`, `new.params` (parameters), `new.state` (initial condition), `ensemble.id` (ensemble ids), `start.time`, and `stop.time`.
 
-The restart functionality is developed using model specific functions by called write_restart.modelname. You need to make sure first that this function is already exist for your desired model.
+The restart functionality is developed using model specific functions by called `write_restart.modelname`. You need to make sure first that this function is already exist for your desired model.
 
 Note: if the ensemble size is set to 1, PEcAn will select the **posterior median** parameter values rather than taking a single random draw from the posterior
 
+This information is currently used by the following PEcAn workflow functions:
 
-## Sensitivity Runs
-
-Only if this section is defined a sensitivity analysis is done. This section will have `<quantile>` or `<sigma>` nodes. If neither are given, the default is to use the median +/- [1 2 3] x sigma (e.g. the 0.00135 0.0228 0.159 0.5 0.841 0.977 0.999 quantiles); If the 0.5 (median) quantile is omitted, it will be added in the code.
+- `write.configs` - ???
+- `write.ensemble.configs` - ???
+- `run.ensemble.analysis()` - ???
```

Comment:
Why the ???

Also, it might be worth, in the discussion above, distinguishing tags used in set up vs those used in post-hoc visualization/analysis. `<variable>` falls into the latter category, and people could easily misinterpret that they need to run a NEW ensemble if they want to run at a different variable. Didn't make it into this release, because Chaitanya flaked on us, but I really think we should deprecate run.ensemble.analysis (and the variables tag) and just handle ensemble visualization within the interactive visualization tools.

Reply:
`???` indicate I wanted to provide a description of those functions, but didn't get to it. I filled these in partially in 4c6d73188, but they should be fleshed out more at a later date.